docs/learn/calls.md

```diff
@@ -283,13 +320,131 @@ In this example, your IDE will provide proper autocompletion and type checking f
 
 ## Best Practices
 
-- Provider Flexibility: Design functions to be provider-agnostic, allowing easy switching between different LLM providers for comparison or fallback strategies.
-- Provider-Specific Prompts: Tailor prompts to leverage unique features of specific providers when needed, using provider-specific `call` decorators. For example, Anthropic is known to handle prompts with XML particularly well.
-- Error Handling: Implement robust error handling to manage API failures, timeouts, or content policy violations, ensuring your application's resilience. Check out our documentation on [using Tenacity](../integrations/tenacity.md) for how to easily add retries to your error handling logic.
-- Streaming for Long Tasks: Utilize streaming for long-running tasks or when providing real-time updates to users, improving the user experience. Check out the [`Streams`](./streams.md) documentation for more details.
-- Modular Function Design: Break down complex tasks into smaller, modular functions that can be chained together, enhancing reusability and maintainability. Check out the [`Chaining`](./chaining.md) documentation for more details.
-- Custom Clients: Leverage custom clients when working with self-hosted models or specific API configurations, allowing for greater flexibility in deployment scenarios. For example, you can use open-source models by serving OpenAI compatible endpoints (e.g. with `vLLM`, `Ollama`, etc).
-- Structured Outputs: Use `response_models` to ensure consistency and type safety when you need structured data from your LLM calls. Check out the [`Response Models`](./response_models.md) documentation for more details.
-- Parameterized Calls: Take advantage of `call_params` to customoize model behavior without changing your core function logic.
-
-Mastering the `call` decorator is the next step towards building robust LLM applications with Mirascope that are flexible, efficient, and adaptable to various providers and use cases.
+To make the most of Mirascope's `call` decorator and ensure you're building robust, efficient, and maintainable LLM applications, consider the following best practices:
+
+- **Provider Flexibility**: Design functions to be provider-agnostic, allowing easy switching between different LLM providers for comparison or fallback strategies.
+   ```python
+   @prompt_template("Recommend a {genre} book")
+   def recommend_book(genre: str):
+       ...
+
+   # Can easily switch between providers
+   openai_response = recommend_book.run(openai.call("gpt-4o-mini"))("fantasy")
```

Comment:
the method doesn't have a `run` method, it should instead be `openai.call("gpt-4o-mini")(recommend_book)("fantasy")`

same for the anthropic line below